### PR TITLE
[hevce] Describe extended QP ranges in the manual

### LINKED ISSUE #1985: Describe extended QP ranges in the manual
See #1980
It looks currently MSDK manual doesn't mention it.

### PR DESCRIPTION
**Pull request by yskuridi**
Fixes #1985.

### CODE REVIEW

**Review by dmitryermilov — COMMENTED** [review]
Cool. Thank you.
Shall we add info about 12 bit ?

File: doc/mediasdk-man.md [hunk 1]
```diff
@@ -3982,7 +3982,7 @@ The application can attach this extended buffer to the [mfxVideoParam](#mfxVideo
...
 `IntRefCycleSize` | Specifies number of pictures within refresh cycle starting from 2. 0 and 1 are invalid values. This parameter is valid only during initialization
-`IntRefQPDelta` | Specifies QP difference for inserted intra MBs. This is signed value in [-51, 51] range. This parameter is valid during initialization and runtime.
+`IntRefQPDelta` | Specifies QP difference for inserted intra MBs. This is signed value in [-51, 51] range if target encoding bit-depth for luma samples is 8 and this range is [-63, 63] for 10 bit-depth. This parameter is valid during initialization and runtime.
```
Comment (dmitryermilov) [inline doc/mediasdk-man.md:3985]: shall we add info about 12 bit ?

File: doc/mediasdk-man.md [hunk 2]
```diff
@@ -6407,7 +6407,7 @@ The **mfxQPandMode** structure specifies specifies per-MB or per-CU mode and QP
 
 | | |
 --- | ---
-`QP` | QP for MB or CU. Valid when Mode = MFX_MBQP_MODE_QP_VALUE.<br>For AVC valid range is 1..51.<br>For HEVC valid range is 1..51. Application’s provided QP values should be valid; otherwise invalid QP values may cause undefined behavior.<br>MBQP map should be aligned for 16x16 block size. (align rule is (width +15 /16) && (height +15 /16))<br>For MPEG2 QP corresponds to quantizer_scale of the ISO*\/IEC* 13818-2 specification and have valid range 1..112.
+`QP` | QP for MB or CU. Valid when Mode = MFX_MBQP_MODE_QP_VALUE.<br>For AVC valid range is 1..51.<br>For HEVC valid range is 1..51 inclusive if target encoding bit-depth for luma samples is 8 and this range is 1..63 for 10 bit-depth. Application’s provided QP values should be valid; otherwise invalid QP values may cause undefined behavior.<br>MBQP map should be aligned for 16x16 block size. (align rule is (width +15 /16) && (height +15 /16))<br>For MPEG2 QP corresponds to quantizer_scale of the ISO*\/IEC* 13818-2 specification and have valid range 1..112.
```
Comment (dmitryermilov) [inline doc/mediasdk-man.md:6410]: same

File: doc/mediasdk-man.md [hunk 3]
```diff
@@ -6444,7 +6444,7 @@ The **mfxExtMBQP** structure specifies per-macroblock QP for current frame if [m
...
+`QP` | Pointer to a list of per-macroblock QP in raster scan order. In case of interlaced encoding the first half of QP array affects top field and the second – bottom field.<br><br>For AVC valid range is 1..51.<br><br>For HEVC valid range is 1..51 inclusive if target encoding bit-depth for luma samples is 8 and this range is 1..63 for 10 bit-depth. Application’s provided QP values should be valid; otherwise invalid QP values may cause undefined behavior. MBQP map should be aligned for 16x16 block size. (align rule is (width +15 /16) && (height +15 /16))<br><br>For MPEG2 QP corresponds to quantizer_scale of the ISO*/IEC* 13818-2 specification and have valid range 1..112.
```
Comment (dmitryermilov) [inline doc/mediasdk-man.md:6447]: same

**Review by dmitryermilov — APPROVED** [review]

**Review by vilichev — APPROVED** [review]

**Comment by lakulako** [comment]
As I remember hevc vdenc has low QP limit of 10. And probably +6*(bits-8). Should it be checked and noted here too?

**Review by lakulako — COMMENTED** [review]
As I remember hevc vdenc has low QP limit of 10. And probably +6*(bits-8). Should it be checked and noted here too?

**Comment by yskuridi** [comment]
> As I remember hevc vdenc has low QP limit of 10. And probably +6*(bits-8). Should it be checked and noted here too?

"Not all codecs and SDK implementations support this value. Use Query function to check if this feature is supported." 
Does that mean that some values from the range could be unsupported for some hw?

**Comment by dmitryermilov** [comment]
> ber hevc vdenc has low QP limit of 10. And probably +6*(bits-8).

It's too low level Gen specific. We even won't be able to port it in VPL spec.

**Pull request merged**